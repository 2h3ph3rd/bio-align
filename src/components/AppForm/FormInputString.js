import * as React from 'react'
import TextField from '@mui/material/TextField'
import InputAdornment from '@mui/material/InputAdornment'
import FormControl from '@mui/material/FormControl'
import EditIcon from '@mui/icons-material/Edit'

/*
 * props: label, defaultValue, handleChange
 */
export default function FormInputString(props) {
  const [value, setValue] = React.useState(props.default)

  const handle = (e) => {
    const value = e.target.value.toUpperCase()
    setValue(value)
    if (!isError(value)) {
      props.handle(value)
    }
  }

  const isError = (value) => {
    return value.length > 8
  }

  return (
    <FormControl fullWidth margin="normal">
      <TextField
        onChange={handle}
        label={props.label}
        value={value}
        variant="outlined"
        error={isError(value)}
        helperText={isError(value) ? 'Max 8 alphabetic characters' : ''}
        data-testid="value-input"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <EditIcon />
            </InputAdornment>
          ),
        }}
      />
    </FormControl>
  )
}
