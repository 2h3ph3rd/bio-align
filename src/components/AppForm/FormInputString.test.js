import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import FormInputString from './FormInputString'

const fakeHandleChange = () => {}
const fakeLabel = 'test'
const fakeDefaultValue = 'TEST'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <FormInputString
        default={fakeDefaultValue}
        label={fakeLabel}
        handle={fakeHandleChange}
      ></FormInputString>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('default value param', () => {
  render(
    <FormInputString
      default={fakeDefaultValue}
      label={fakeLabel}
      handle={fakeHandleChange}
    ></FormInputString>
  )

  const input = screen.getByTestId('value-input').querySelector('input')
  expect(input).toHaveValue(fakeDefaultValue)
})

test('handle change param', () => {
  const newValue = 'TEST'
  const fakeHandleChange = (value) => {
    expect(value).toBe(newValue)
  }
  render(
    <FormInputString
      default={fakeDefaultValue}
      label={fakeLabel}
      handle={fakeHandleChange}
    ></FormInputString>
  )

  const input = screen.getByTestId('value-input').querySelector('input')
  expect(input).toBeInTheDocument()

  fireEvent.change(input, { target: { value: newValue } })
  expect(input).toHaveValue(newValue)
})

test('shuffle', () => {
  render(
    <FormInputString
      default={fakeDefaultValue}
      label={fakeLabel}
      handle={fakeHandleChange}
    ></FormInputString>
  )

  const button = screen.getByTestId('shuffle-button')
  expect(button).toBeInTheDocument()

  fireEvent.click(button)

  const input = screen.getByTestId('value-input').querySelector('input')
  expect(input).toBeInTheDocument()

  expect(input.value).not.toEqual(fakeDefaultValue)
})
