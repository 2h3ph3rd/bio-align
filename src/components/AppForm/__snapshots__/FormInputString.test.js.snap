// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    data-testid="value-input"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
    >
      test
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="ACTC"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1z7n62"
        >
          <span>
            test
          </span>
        </legend>
      </fieldset>
    </div>
    
  </div>
</div>
`;
